Move from @download/blockies to blo

bpierre opened the pull request:
## Description

Move from @download/blockies to [blo](https://github.com/bpierre/blo), for a faster and more precise rendering of the blockies identicons, while also removing the need for the library to access `document`.

## **Manual testing steps**

1. `yarn && yarn start`
2. In Settings => General, change the identicon style to “Blockies”
3. Check everywhere there is an identicon in the extension.

## **Screenshots/Recordings**

Before:
<img src=https://github.com/MetaMask/metamask-extension/assets/36158/e4a91c97-87bc-45fc-8f08-9514bb6dca8f width=400>

After:
<img src=https://github.com/MetaMask/metamask-extension/assets/36158/18e80f1f-3775-4457-a2a2-ce3915b5b6c7 width=400>

Before:
<img src=https://github.com/MetaMask/metamask-extension/assets/36158/763c333b-17f0-464a-a5b1-33813a14c7df width=600>

After:
<img src=https://github.com/MetaMask/metamask-extension/assets/36158/a9aad236-0a23-42cb-a15a-61a262cba777 width=600>

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable
- [x] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

bpierre committed:
Move from @download/blockies to blo

bpierre commented on lavamoat/browserify/beta/policy.json:
```diff
@@ -10,11 +10,6 @@
         "regeneratorRuntime": "write"
       }
     },
-    "@download/blockies": {
-      "globals": {
-        "document.createElement": true
-      }
-    },
```
I don’t know if this is generated or not, but I removed these from all the policy.json files. Let me know if there is another way to do it.

brad-decker replied:
The policy files are generated by running `yarn lavamoat:auto` but we can also use our bot to update them automatically so hold tight :)

bpierre replied:
Cool :)

bpierre replied:
Thanks :pray:

bpierre commented:
I have read the CLA Document and I hereby sign the CLA

brad-decker commented:
@metamaskbot update-policies

bpierre committed:
Fix account.picker.test.js

brad-decker commented:
Reviewers, check out https://github.com/bpierre/blo author is the publisher of the new dependency.

bpierre committed:
Update jest snapshot

bpierre committed:
ESLint: allow importing .cjs files

bpierre commented on .eslintrc.js:
```diff
@@ -161,6 +161,8 @@ module.exports = {
             // Always try to resolve types under `<root>/@types` directory even
             // it doesn't contain any source code, like `@types/unist`
             alwaysTryTypes: true,
+            // Allow importing .cjs files
+            extensions: ['.js', '.cjs'],
```
Let me know if this is something you’d rather not change in your ESLint config, if that’s the case I can publish a new version of the package with `.cjs.js` for the CJS export.

legobeat committed:
update lavamoat browserify policies

legobeat committed:
Merge branch 'develop' into blo

naugtur reviewed (commented):
I've reviewed the blo package and it doesn't contain anything suspicious.

A dependency introduced by someone from the community should be pinned to the reviewed version. (I mean Pierre looks super trustworthy, but as a rule in situations like this - if it's a plot to get an attack in, the attack is coming in an update)

Thanks for your contribution @bpierre

brad-decker commented:
@bpierre could you fix the conflicts and fix the version (remove the ^) :) thank you

bpierre committed:
Dependencies: pin blo to 1.1.0

bpierre committed:
Merge branch develop into blo

bpierre commented:
@brad-decker @naugtur done and done :)

bpierre committed:
Merge branch develop into blo

brad-decker commented:
@SocketSecurity ignore blo@1.1.0

brad-decker reviewed (dismissed):
LGTM. @bpierre i think it makes sense to add the other format and we can comeback and remove the additional extension from the eslint config -- but that can be done later. We'll do a review of the library for every release to see what has changed so if possible keep releases well documented and segmented (but the repo looks good atm 👍🏻) Thanks for this contribution and excellent work.

bpierre commented:
@brad-decker awesome, thank you! :pray: I’ll publish a new release with a .js extension for the CJS export and let you know.

> if possible keep releases well documented and segmented

That’s the plan :slightly_smiling_face: :+1:

bpierre committed:
Merge branch 'develop' into blo

bpierre commented:
I published [`blo@1.1.1`](https://github.com/bpierre/blo/releases/tag/v1.1.1) which uses `.cjs.js` for the CJS extension.

Diff: https://diff.intrinsic.com/blo/1.1.0/1.1.1

Should I change the blo version in this PR?

bpierre committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

legobeat committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

brad-decker commented:
@bpierre lets update it here :)

bpierre committed:
Upgrade blo to 1.1.1

This blo version provides a CJS export using a .js extension rather than
.cjs, removing the need to add an extra ESLint setting.

bpierre committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'blo' of github.com:bpierre/metamask-extension into blo

bpierre commented:
I weirdly couldn’t reproduce the ESLint error when using 1.1.0 and removing the config line :thinking:

I updated blo to 1.1.1 anyway, and removed the change from the ESLint config :+1:

bpierre committed:
Merge branch 'develop' into blo

legobeat committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

brad-decker committed:
Merge branch 'develop' into blo

legobeat committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

bpierre commented:
@brad-decker no problem at all! I updated the branch, let’s see if the test suite passes.

bpierre committed:
Merge branch 'develop' into blo

bpierre committed:
Merge branch 'develop' into blo

legobeat committed:
Merge branch 'develop' into blo

legobeat committed:
Merge branch 'develop' into blo
